harvester/harvest/alma.py

```diff
@@ -0,0 +1,192 @@
+"""harvester.harvest.alma"""
+
+import glob
+import logging
+import re
+from collections.abc import Iterator
+from typing import Literal, cast
+
+import smart_open  # type: ignore[import-untyped]
+from attrs import define, field
+from dateutil.parser import parse as date_parser
+from lxml import etree
+from marcalyx import Record as MARCRecord  # type: ignore[import-untyped]
+
+from harvester.aws.s3 import S3Client
+from harvester.config import Config
+from harvester.harvest import Harvester
+from harvester.records import (
+    MARC,
+    Record,
+)
+from harvester.utils import convert_to_utc
+
+logger = logging.getLogger(__name__)
+
+CONFIG = Config()
+
+
+@define
+class MITAlmaHarvester(Harvester):
+    """Harvester of MIT Alma MARC Records."""
+
+    input_files: str = field(default=None)
+
+    def full_harvest_get_source_records(self) -> Iterator[Record]:
+        """Identify files for harvest by parsing MARC records from FULL Alma exports."""
+        CONFIG.check_required_env_vars()
+        yield from self._get_source_records()
+
+    def incremental_harvest_get_source_records(self) -> Iterator[Record]:
+        """Identify files for harvest by parsing MARC records from DAILY Alma exports."""
+        CONFIG.check_required_env_vars()
+        yield from self._get_source_records()
+
+    def _get_source_records(self) -> Iterator[Record]:
```

Comment:
What is the difference between a full and incremental harvest for Alma?

Reply:
At this time, the only difference is the `Harvester.harvest_type` property that is set.  When the flag is `full`, then we look for `full` in the S3 filepaths.  When flag is `incremental`, then we look for `daily` in the S3 filepath.

This harvester is a bit more streamlined vs MIT and OGM, as the files are in the same location, and have the same structure.  The only difference is identifying _which_ files to include.  Hence, the date filtering as well.